### ISSUE TITLE
Async linker

### ISSUE THREAD

**Issue by stefnotch** [stefnotch/wgsl_linker#1]
It'd be nice to have an async version of the linker. 
- [ ] Asynchronous compilation
  - [ ] Fetch missing files in parallel
- [ ] Multiple compilations running in parallel

I don't yet know what exactly the use cases are, and what the API should look like.

**Comment by stefnotch** [comment]
Superseded by wesl-rs
